state/validators.md

```diff
@@ -0,0 +1,40 @@
+# Sequencer Selection Scheme
+
+## Abstract
+
+The Sequencer Selection scheme describes the process of selecting a block proposer i.e. sequencer from the validator set.
+
+## Protocol/Component Description
+
+There is exactly one sequencer which is configured at genesis. `GenesisDoc` usually contains an array of validators as it is imported from `CometBFT`. If there is more than one validator defined
+in the genesis validator set, an error is thrown.
+
+The `State` struct defines a `Sequencer` field which contains the public key of the first and only validator of the genesis.
+
+The `Header` struct defines a field called `ProposerAddress` which is the pubkey of the original proposer of the block.
+
+The `SignedHeader` struct commits over the header and the proposer address and stores the result in `LastCommitHash`.
+
+A new untrusted header is verified by checking its `ProposerAddress` and matching it against the best-known header. In case of a mismatch, an error is thrown.
```

Comment:
Ideally in vA, i want to check the ProposerAddress against the genesis, not the best-known header. It's more efficient, as you don't have to wait for the previous to arrive. It's possible because we won't have a way to change the sequencer without a hard-fork.

Reply:
go-header might make this difficult to implement. i sent [a PR](https://github.com/celestiaorg/go-header/pull/128) but it's not gonna get merged. so this is up in the air, i'm going to try to hack it with `Verify` on non-adjacent headers. if that doesn't work, then we'll have to do it with best known header.

Maybe we could write the spec this way for now? LMK your thoughts, otherwise lgtm

Reply:
Ok, we will merge it as is and consider updating the spec after these issues are resolved
https://github.com/rollkit/rollkit/issues/1273
https://github.com/rollkit/rollkit/issues/1272
https://github.com/rollkit/rollkit/issues/1270